pkg/services/star/model.go

```diff
@@ -41,14 +43,17 @@ func (cmd *UnstarDashboardCommand) Validate() error {
 // QUERIES
 
 type GetUserStarsQuery struct {
+	OrgID  int64 `xorm:"-"`
 	UserID int64 `xorm:"user_id"`
 }
 
 type IsStarredByUserQuery struct {
-	UserID      int64 `xorm:"user_id"`
-	DashboardID int64 `xorm:"dashboard_id"`
+	UserID       int64  `xorm:"user_id"`
+	DashboardID  int64  `xorm:"dashboard_id"`
+	DashboardUID string `xorm:"-"`
 }
 
 type GetUserStarsResult struct {
-	UserStars map[int64]bool
+	UserStars    map[int64]bool
+	UserStarsUID []string `json:"userStarsUid"`
```

Comment:
Since this is a "breaking" change... and what is returned in a result, can we change the model to:
```suggestion
	Dashboards []string `json:"dashboard"`
```

On the frontend, we will stick it in a `Set()` and check if items are starred there.

Reply:
`DashboardUIDs` would be a better name than `UserStarsUID`, I'll change to that.

The API is supposed to return the same thing as it previously did, I'm moving the logic to create the list of dashboards into the service and reducing the number of calls to the database from `O(n)` to `O(1)`.

```
	uids := []string{}
 	for dashboardId := range iuserstars.UserStars {
 		query := &models.GetDashboardQuery{
 			Id:    dashboardId,
 			OrgId: c.OrgID,
 		}
 		err := hs.DashboardService.GetDashboard(c.Req.Context(), query)

 		// Grafana admin users may have starred dashboards in multiple orgs.  This will avoid returning errors when the dashboard is in another org
 		if err == nil {
 			uids = append(uids, query.Result.Uid)
 		}
 	}
 	return response.JSON(200, uids)
```

Reply:
ah I see. 👍 

I talked to @ying-jeanne about using this as a simple example to try with object store also.  But the API still need to accept uids

Reply:
Can we deprecate the id path?  at least make it clear that is going away and is internal?